### PR TITLE
Update Developer Guide re. tab completion (#9918)

### LINKED ISSUE #9918: Py3 Python console: tab completion no longer shows private attributes
### Is your feature request related to a problem? Please describe.
In python 2, pressing tab in the python console gives us auto completion for an object. For example, typing `nav.` followed with tab gives us all the properties on the cached navigator object, including the ones starting with one or two underscores. In python 3 however, these attributes are hidden.

### Describe the solution you'd like
I think I like this new approach, but it should be documented as a change in the section of changes for developers.

### Describe alternatives you've considered
Restore the python 2 behavior.

### PR DESCRIPTION
**Pull request by JulienCochuyt**
### Link to issue number:

Fixes #9918

### Summary of the issue:

In the Python console, the tab-completion is smarter with Python 3.

### Description of how this pull request fixes the issue:

Document the new behavior in the Developer Guide.

### Testing performed:

None.

### Change log entry:

Section: Changes for developpers

The tab-completion in the Python console is now smarter. Refer to the Developer Guide for more details.

### CODE REVIEW

File: developerGuide.t2t [hunk 1]
```diff
@@ -796,6 +796,15 @@ These variables are:
 - brlRegions: The braille regions from the active braille buffer
 -
 
+++ Tab completion ++
+The input control supports tab-completion of variables and member attributes names.
+Hit the tab key once to complete the current input if there is one single candidate.
+If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
```
Comment (XLTechie) [inline developerGuide.t2t]: I think that should be "attribute names"

File: developerGuide.t2t [hunk 2]
```diff
@@ -796,6 +796,15 @@ These variables are:
...
 -
 
+++ Tab completion ++
+The input control supports tab-completion of variables and member attributes names.
+Hit the tab key once to complete the current input if there is one single candidate.
+If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
+If the input is "nav._", attributes names with a single leading underscore are proposed.
+Similarily, if the input is "nav.__", attributes names with two leading underscores are proposed.
```
Comment (XLTechie) [inline developerGuide.t2t]: S/b "Similarly"

Comment (XLTechie) [inline developerGuide.t2t]: All three of those s/b "attribute names" IMO.

```suggestion
That is, if the input is "nav.", attributes names with no leading underscore are proposed.
If the input is "nav._", attributes names with a single leading underscore are proposed.
Similarily, if the input is "nav.__", attributes names with two leading underscores are proposed.
```
```suggestion
That is, if the input is "nav.", attribute names with no leading underscore are proposed.
If the input is "nav._", attribute names with a single leading underscore are proposed.
Similarly, if the input is "nav.__", attribute names with two leading underscores are proposed.
```

**Comment by XLTechie** [comment]
Apologies, that suggested change did not work as intended to include multiple 
lines in the diff.

I will delete it if it lets me.

The idea was to suggest changing all three "attributes names" to "attribute 
names", and "similarily" to "similarly".

File: developerGuide.t2t (hunk 1 above)
Comment (XLTechie) [inline developerGuide.t2t]:
```suggestion
That is, if the input is "nav.", attribute names with no leading underscore are proposed.
```

File: developerGuide.t2t [hunk 3]
```diff
@@ -796,6 +796,15 @@ These variables are:
 - brlRegions: The braille regions from the active braille buffer
 -
 
+++ Tab completion ++
+The input control supports tab-completion of variables and member attributes names.
+Hit the tab key once to complete the current input if there is one single candidate.
+If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
+If the input is "nav._", attributes names with a single leading underscore are proposed.
```
Comment (XLTechie) [inline developerGuide.t2t]:
```suggestion
If the input is "nav._", attribute names with a single leading underscore are proposed.
```

**Comment by JulienCochuyt** [comment]
@XLTechie, thank you for the feedback: I'm not a native English speaker.

**Comment by XLTechie** [comment]
Sorry for the overlapping inline comments. Your "fix-up" commit wasn't showing 
when I posted them, but the notification for it came in before the notifications 
for my corrected inlines.
Not sure what the deal was there, but sorry about all the confusion.

File: developerGuide.t2t [hunk 4]
```diff
@@ -801,9 +801,9 @@ The input control supports tab-completion of variables and member attributes nam
 Hit the tab key once to complete the current input if there is one single candidate.
 If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
```
Comment (XLTechie) [inline developerGuide.t2t]: I Missed this one, although it is not really important.

```suggestion
If there is more than one, hit the tab key a second time to open a menu listing all matching possibilities.
```

**Comment by JulienCochuyt** [comment]
> Sorry for the overlapping inline comments

No problem, feel free to further improve the proposed phrasing.

**Comment by XLTechie** [comment]
I don't see anything else to improve.:) The only important one was 
"similarly"--all the rest were just me being a pedantic grammarian.

**Review by feerrenrut — APPROVED** [review]
Thanks @JulienCochuyt

**Pull request merged**